src/modules/ffmpeg/producer/av_producer.cpp

```diff
@@ -104,8 +104,12 @@ struct Decoder
 
         FF(av_opt_set_int(ctx.get(), "refcounted_frames", 1, 0));
 
-        // TODO (fix): Remove limit.
-        FF(av_opt_set_int(ctx.get(), "threads", env::properties().get(L"configuration.ffmpeg.producer.threads", 4), 0));
+        int defaultNumThreads =
+            codec->capabilities & AV_CODEC_CAP_AUTO_THREADS
+                ? 0 : codec->capabilities & (AV_CODEC_CAP_FRAME_THREADS | AV_CODEC_CAP_SLICE_THREADS)
+                        ? std::thread::hardware_concurrency()
+                        : 1;
+        FF(av_opt_set_int(ctx.get(), "threads", env::properties().get(L"configuration.ffmpeg.producer.threads", defaultNumThreads), 0));
```

Comment:
I think it's better to just simply set it to "auto".

Reply:
Thanks Robert.
I was just writing up some notes about this in the issue: https://github.com/CasparCG/server/issues/1281#issuecomment-644859100

Reply:
Sounds good!